src/webgl/p5.RendererGL.js

```diff
@@ -1899,44 +2011,100 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
+    const envCubemap = this.GL.createTexture();
+    this.GL.bindTexture(this.GL.TEXTURE_CUBE_MAP, envCubemap);
+
+    // Loop through each face and allocate storage
+    for (let i = 0; i < 6; ++i) {
+      this.GL.texImage2D(
+        this.GL.TEXTURE_CUBE_MAP_POSITIVE_X + i, 0,
+        this.GL.RGB16F, 512, 512, 0,
+        this.GL.RGB, this.GL.FLOAT, null
       );
     }
-    newFramebuffer.draw(() => {
-      this._pInst.shader(this.diffusedShader);
-      this.diffusedShader.setUniform('environmentMap', input);
-      this._pInst.noStroke();
-      this._pInst.rectMode(constants.CENTER);
-      this._pInst.noLights();
-      this._pInst.rect(0, 0, width, height);
+
+    // Set parameters for the cubemap
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_WRAP_S, this.GL.CLAMP_TO_EDGE);
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_WRAP_T, this.GL.CLAMP_TO_EDGE);
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_WRAP_R, this.GL.CLAMP_TO_EDGE);
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_MIN_FILTER, this.GL.LINEAR);
+    this.GL.texParameteri(this.GL.TEXTURE_CUBE_MAP
+      , this.GL.TEXTURE_MAG_FILTER, this.GL.LINEAR);
+
+    // Set up view matrices for each face
+    const captureViews = [
+      lookAt([0, 0, 0], [1, 0, 0], [0, -1, 0]), // Positive X
+      lookAt([0, 0, 0], [-1, 0, 0], [0, -1, 0]), // Negative X
+      lookAt([0, 0, 0], [0, 1, 0], [0, 0, 1]), // Positive Y
+      lookAt([0, 0, 0], [0, -1, 0], [0, 0, -1]), // Negative Y
+      lookAt([0, 0, 0], [0, 0, 1], [0, -1, 0]), // Positive Z
+      lookAt([0, 0, 0], [0, 0, -1], [0, -1, 0]) // Negative Z
+    ];
+
+    // Get the cubemap shader
+    const cubemapShader = this._getCubemapShader();
+    this._pInst.shader(cubemapShader);
+    cubemapShader.setUniform('equirectangularMap', 0);
+
+    // Define captureProjection as a 4x4 projection matrix
+    let captureProjection = new p5.Matrix();
+    captureProjection.perspective(Math.PI / 2.0, 1.0, 0.1, 10.0);
+
+    cubemapShader.setUniform('projection', captureProjection.mat4);
+
+    this.GL.activeTexture(this.GL.TEXTURE0);
+    this.GL.bindTexture(this.GL.TEXTURE_2D, input);
+
+    // Create a new framebuffer
+    let newFramebuffer = this._pInst.createFramebuffer({
+      width: 512,
+      height: 512,
+      density: 1
     });
-    this.diffusedTextures.set(input, newFramebuffer);
-    return newFramebuffer;
+
+    this.GL.bindFramebuffer(this.GL.FRAMEBUFFER, newFramebuffer.handle);
+
+    // Render each face of the cubemap
+    for (let i = 0; i < 6; ++i) {
+      this.GL.framebufferTexture2D(
+        this.GL.FRAMEBUFFER,
+        this.GL.COLOR_ATTACHMENT0,
+        this.GL.TEXTURE_CUBE_MAP_POSITIVE_X + i,
+        envCubemap,
+        0
+      );
+
+      this.GL.clear(this.GL.COLOR_BUFFER_BIT | this.GL.DEPTH_BUFFER_BIT);
+
+      cubemapShader.setUniform('view', captureViews[i].mat4);
```

Comment:
Same with this. We should not calculate the view matrix. We already had set up the modelVeiw matrix as "uModelViewMatrix".